craft_parts/sequencer.py

```diff
@@ -273,10 +276,21 @@ def _rerun_step(
         self._run_step(part, step, reason=reason, rerun=True)
 
     def _update_step(self, part: Part, step: Step, *, reason: Optional[str] = None):
+        """Set the step state as reexecuted by updating its timestamp."""
         logger.debug("update step %s:%s", part.name, step)
         self._add_action(part, step, action_type=ActionType.UPDATE, reason=reason)
         self._sm.update_state_timestamp(part, step)
 
+    def _reapply_layer(
+        self, part: Part, layer_hash: LayerHash, *, reason: Optional[str] = None
+    ):
+        """Update the layer hash without changing the step state."""
+        logger.debug("reapply layer %s: hash=%s", part.name, layer_hash)
+        self._layer_state.set_layer_hash(part, layer_hash)
+        self._add_action(
+            part, Step.OVERLAY, action_type=ActionType.REAPPLY, reason=reason
+        )
```

Comment:
This is the layer reapplication during planning. It only changes state in memory and doesn't change the filesystem.